### DISCUSSION TITLE
v4.0.1

### DISCUSSION THREAD

**Discussion by klaasnicolaas** [klaasnicolaas/home-assistant-glow#430] [Announcements]
## What's changed

A release full of improvements in the documentation and package imports.

./Klaas

## 🐛 Bug fixes

- #374 BUG fix:  Added properly top case to the cases' folder @knaak
- #402 Fix the package import URLs for d1_mini and esp8266 @klaasnicolaas
- #420 Include full YAML config after dashboard import @klaasnicolaas
- #427 Fix issue with package import during build @klaasnicolaas

## 🚀 Enhancements

- #375 Remove because is is also the default setting @klaasnicolaas
- #420 Include full YAML config after dashboard import @klaasnicolaas

## 🧰 Maintenance

- #376 Fix issues in github actions @klaasnicolaas
- #396 Fix skipping because of empty matrix @klaasnicolaas

## 📚 Documentation

- #368 contribute an esp32-wroom-32d case design @knaak
- #372 Add Google Analytics @klaasnicolaas
- #376 Fix issues in github actions @klaasnicolaas
- #377 Add github as icon in the navbar @klaasnicolaas
- #392 Add FAQ item about installation process (klaasnicolaas#274) @ahmadyusuf
- #399 Rewording the firmware installation instruction @klaasnicolaas
- #418 Update the FAQ for using the filter from pulse_meter @klaasnicolaas
- #419 Add links to case files on Printables @klaasnicolaas
- #428 Update ESPHome firmware customization documentation @klaasnicolaas

## ⬆️ Dependency updates

<details>
<summary>34 changes</summary>

- #364 ⬆️ Lock file maintenance @renovate
- #366 ⬆️ Lock file maintenance @renovate
- #367 ⬆️ Lock file maintenance @renovate
- #369 ⬆️ Update actions/upload-artifact action to v4.2.0 @renovate
- #370 ⬆️ Lock file maintenance @renovate
- #371 ⬆️ Update actions/upload-artifact action to v4.3.0 @renovate
- #379 ⬆️ Update dependency @docusaurus/module-type-aliases to v3.1.1 @renovate
- #380 ⬆️ Update dependency @docusaurus/preset-classic to v3.1.1 @renovate
- #382 ⬆️ Update dependency @docusaurus/types to v3.1.1 @renovate
- #381 ⬆️ Update dependency @docusaurus/tsconfig to v3.1.1 @renovate
- #378 ⬆️ Update dependency @docusaurus/core to v3.1.1 @renovate
- #385 ⬆️ Update release-drafter/release-drafter action to v6 @renovate
- #386 ⬆️ Lock file maintenance @renovate
- #387 ⬆️ Update actions/deploy-pages action to v4.0.4 @renovate
- #389 ⬆️ Update actions/upload-artifact action to v4.3.1 @renovate
- #390 ⬆️ Update actions/setup-node action to v4.0.2 @renovate
- #391 ⬆️ Update actions/upload-pages-artifact action to v3.0.1 @renovate
- #388 ⬆️ Update actions/download-artifact action to v4.1.2 @renovate
- #393 ⬆️ Lock file maintenance @renovate
- #397 ⬆️ Update dependency @mdx-js/react to v3.0.1 @renovate
- #398 ⬆️ Update dependency esp-web-tools to v10 @renovate
- #400 ⬆️ Lock file maintenance @renovate
- #404 ⬆️ Lock file maintenance @renovate
- #403 ⬆️ Update esphome/build-action action to v1.9.0 @renovate
- #405 ⬆️ Update actions/download-artifact action to v4.1.3 @renovate
- #408 ⬆️ Update actions/download-artifact action to v4.1.4 @renovate
- #410 ⬆️ Lock file maintenance @renovate
- #415 ⬆️ Update dependency typescript to ~5.4.0 @renovate
- #416 ⬆️ Lock file maintenance @renovate
- #407 ⬆️ Update esphome/build-action action to v2 @renovate
- #422 ⬆️ Lock file maintenance @renovate
- #424 ⬆️ Update dependency typescript to v5.4.3 @renovate
- #423 ⬆️ Update actions/deploy-pages action to v4.0.5 @renovate
- #429 Bump webpack-dev-middleware from 5.3.3 to 5.3.4 in /docs @dependabot
</details>

**Full Changelog**: https://github.com/klaasnicolaas/home-assistant-glow/compare/v4.0.0...v4.0.1

<hr /><em>This discussion was created from the release <a href='https://github.com/klaasnicolaas/home-assistant-glow/releases/tag/v4.0.1'>v4.0.1</a>.</em>

**Comment by phlexss** [comment]
I have installed v4.0.1 and adopted it in ESPhome in HA.....now when in ESPhome I click "Edit" in  home-assistant-glow-xxxx I don't see any values that are from pulse_meter.yaml

Am I wrong in assuming that in this release it would now be possible to edit everything in the ESPhome dashboard once Glow is adopted?

If so, how can I do it?

Thanks!

**Reply by klaasnicolaas** [reply]
That's all in the docs... You can use the `!extend` to modify existing components.

https://glow-energy.io/docs/advanced/firmware_customization#make-customizations

**Reply by phlexss** [reply]
Aha....I missed that one...sorry!

Thanks for pointing that out!
